src/logics/UFLATheory.cc

```diff
@@ -8,25 +8,17 @@
 #include "Substitutor.h"
 #include "TreeOps.h"
 
-bool UFLATheory::simplify(const vec<PFRef>& formulas, PartitionManager &, int curr)
-{
-    auto & currentFrame = pfstore[formulas[curr]];
-    if (this->keepPartitions()) {
-        throw OsmtInternalException("Mode not supported for QF_UFLRA yet");
-    } else {
-        PTRef coll_f = getCollateFunction(formulas, curr);
-        PTRef fla = applySubstitutionBasedSimplificationIfEnabled(coll_f);
-        fla = rewriteDistincts(getLogic(), fla);
-        fla = rewriteDivMod<ArithLogic>(logic, fla);
-        PTRef purified = purify(fla);
-        if (logic.hasArrays()) {
-            purified = instantiateReadOverStore(logic, purified);
-        }
-        PTRef noArithmeticEqualities = splitArithmeticEqualities(purified);
-        this->getTSolverHandler().setInterfaceVars(getInterfaceVars(noArithmeticEqualities));
-        currentFrame.root = noArithmeticEqualities;
+PTRef UFLATheory::preprocessAfterSubstitutions(PTRef fla, PreprocessingContext const & context) {
```

Comment:
Is it possible to do before substitutions as well?

Reply:
It is, but this theory was not doing anything before substitutions before, so it is also not doing anything in this PR.